subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -403,21 +454,23 @@ static int cc_rx_data_handler(const struct nct_evt *nct_evt)
 	enum nfsm_state new_state;
 	const struct nrf_cloud_data *payload = &nct_evt->param.cc->data;
 	bool config_found = false;
+	bool control_found = false;
 	const enum nfsm_state current_state = nfsm_get_current_state();
 
 	handle_device_config_update(nct_evt, &config_found);
+	handle_device_control_update(nct_evt, &control_found);
```

Comment:
`handle_device_config_update` and `handle_device_control_update` both do similar things... parsing the input into a cJSON obj and then checking it.  it would be nice if the input could be parsed into onc cJSON obj and checked for cfg and ctrl. 
not sure if there is a clean/simple way to do this.
or this can be considered for a future improvement.

Reply:
future -- see https://projecttools.nordicsemi.no/jira/browse/IRIS-5523